src/OrbitQt/orbithistogram.cpp

```diff
@@ -0,0 +1,167 @@
+constexpr int kTickLength = 5;
+
+[[nodiscard]] static int RoundToClosestInt(double x) { return static_cast<int>(std::round(x)); }
+
+[[nodiscard]] static int Width(const QPainter& painter) { return painter.device()->width(); }
+
+[[nodiscard]] static int WidthMargin(const QPainter& painter) {
+  return RoundToClosestInt(Width(painter) * kRelativeMargin);
+}
+
+[[nodiscard]] static int Height(const QPainter& painter) { return painter.device()->height(); }
+
+[[nodiscard]] static int HeightMargin(const QPainter& painter) {
+  return RoundToClosestInt(Height(painter) * kRelativeMargin);
+}
+
+// if `length > 0`, the line will be plot to the right from `start` and to the left otherwise
+static void DrawHorizontalLine(QPainter& painter, const QPoint& start, int length) {
+  painter.drawLine(start, {start.x() + length, start.y()});
+}
+
+// if `length > 0`, the line will be plot downwards from `start` and upwards otherwise
+static void DrawVerticalLine(QPainter& painter, const QPoint& start, int length) {
+  painter.drawLine(start, {start.x(), start.y() + length});
+}
+
+static void DrawHorizontalAxis(QPainter& painter, const QPoint& zero,
+                               const orbit_statistics::Histogram& histogram, int length) {
+  DrawHorizontalLine(painter, zero, length);
+
+  int tick_spacing_as_value = RoundToClosestInt(static_cast<double>(histogram.max) / kTicksNum);
+  int tick_spacing_pixels = RoundToClosestInt(static_cast<double>(length) / kTicksNum);
+
+  int current_tick_location = tick_spacing_pixels + zero.x();
+  int current_tick_value = tick_spacing_as_value;
+
+  QFontMetrics font_metrics(painter.font());
+
+  for (uint32_t i = 1; i <= kTicksNum; ++i) {
+    DrawVerticalLine(painter, {current_tick_location, zero.y()}, kTickLength);
+
+    QString tick_label = QString::fromStdString(
+        orbit_display_formats::GetDisplayTime(absl::Nanoseconds(current_tick_value)));
+    QRect tick_label_bounding_rect = font_metrics.boundingRect(tick_label);
+    painter.drawText(current_tick_location - tick_label_bounding_rect.width() / 2,
+                     zero.y() + kTickLength + tick_label_bounding_rect.height(), tick_label);
+
+    current_tick_location += tick_spacing_pixels;
+    current_tick_value += tick_spacing_as_value;
+  }
+}
+
+static void DrawVerticalAxis(QPainter& painter, const QPoint& zero, int length, double max_freq) {
+  DrawVerticalLine(painter, zero, -length);
+
+  double tick_spacing_as_value = max_freq / kTicksNum;
+  double current_tick_value = tick_spacing_as_value;
+
+  int tick_spacing_pixels = RoundToClosestInt(static_cast<double>(length) / kTicksNum);
+  int current_tick_location = zero.y() - tick_spacing_pixels;
+
+  QFontMetrics font_metrics(painter.font());
+
+  for (uint32_t i = 1; i <= kTicksNum; ++i) {
+    DrawHorizontalLine(painter, {zero.x(), current_tick_location}, -kTickLength);
+
+    QString tick_label = QString::fromStdString(absl::StrFormat("%.2f", current_tick_value));
```

Comment:
This is not related at all with time, so we shouldn't use DisplayFormats. Consider choosing a different name convention. Why you are using tick?

Reply:
I call them ticks because these short lines on the axes of plots are usually called ticks. It's not necessarily about time. See `matplotlib.pyplot.xticks` for an example.

Reply:
Yeah, but my guess is that @karupayun means the x-axis is not about a timeline. These ticks don't chronologically follow each other, the x-axis is not a timeline.

Reply:
Wait, I'm probably missing something. Aren't this related to the y-axis and therefore only probability and not time at all?

Reply:
D'oh... sorry for the confusion, my bad. I just skimmed over it and mixed up the methods, I thought I was looking at the horizontal axis here. I see you're already using `DisplayFormats` there, please ignore my comments.